### PR TITLE
Degree ordered sorting

### PR DESCRIPTION
**Pull request by foxjas**
Added graph read option DIRECTED_BY_DEGREE, which turns undirected graph into directed by keeping edges with src having higher degree than dst. This is a useful heuristic for reducing max degree.

### CODE REVIEW

File: src/Graph/GraphStd.cpp [hunk 1]
```diff
@@ -228,11 +228,13 @@ void GraphStd<vid_t, eoff_t>::COOtoCSR() noexcept {
         }
         delete[] random_array;
     }
+    /*
```
Comment (kaatish) [inline src/Graph/GraphStd.cpp]: Can the commented out code be removed?

File: src/Graph/GraphStd.cpp [hunk 2]
```diff
@@ -248,6 +250,52 @@ void GraphStd<vid_t, eoff_t>::COOtoCSR() noexcept {
...
+            v = _coo_edges[i].second;
+            deg_u = _out_degrees[u];
+            deg_v = _out_degrees[v];
+            if ((deg_u > deg_v) || ((deg_u == deg_v) && (u > v))) {
+                coo_edges_tmp[counter++] = {u, v}; 
+            }
+        }
+        _coo_edges = coo_edges_tmp;
+        _nE = counter;
+        //std::cout << "Degree-directed graph has " << _nE << " edges" << std::endl;
```
Comment (kaatish) [inline src/Graph/GraphStd.cpp]: Should be removed. If this is a necessary debugging message then perhaps it can encapsulated in a #ifdef

**Comment by foxjas** [comment]
Removed comments according to requested changes. Squashed changes into old commits.

**Comment by kaatish** [comment]
That's great!

**Pull request merged**